packages/template-retail-react-app/app/pages/checkout/partials/shipping-multi-address.jsx

```diff
@@ -0,0 +1,522 @@
+                            </Text>
+                        </ListItem>
+                    )
+                })}
+            {includeQuantity && (
+                <ListItem>
+                    <Text lineHeight={1} color="gray.700" fontSize="sm">
+                        {formatMessage(QUANTITY_LABEL)}: {variant.quantity}
+                    </Text>
+                </ListItem>
+            )}
+        </List>
+    )
+}
+
+MultiShippingItemAttributes.propTypes = {
+    variant: PropTypes.object.isRequired,
+    includeQuantity: PropTypes.bool
+}
+
+const ShippingMultiAddress = ({
+    basket,
+    onSubmit,
+    submitButtonLabel,
+    addNewAddressLabel,
+    noItemsInBasketMessage,
+    deliveryAddressLabel
+}) => {
+    const {formatMessage} = useIntl()
+    const productIds = basket?.productItems?.map((item) => item.productId).join(',')
+    const {
+        data: products,
+        isLoading: productsLoading,
+        error: productsError
+    } = useProducts(
+        {parameters: {ids: productIds, allImages: true}},
+        {enabled: Boolean(productIds)}
+    )
+    const productsMap =
+        products?.data?.reduce((acc, p) => {
+            acc[p.id] = p
+            return acc
+        }, {}) || {}
+    const {data: customer} = useCurrentCustomer()
+    const addresses = customer?.addresses || []
+    const [selectedAddresses, setSelectedAddresses] = useState({})
+
+    // Early return after hooks
+    if (!basket?.productItems?.length) {
+        return (
+            <Box
+                p={8}
+                textAlign="center"
+                color="gray.500"
+                role="status"
+                aria-live="polite"
+                aria-label={formatMessage(noItemsInBasketMessage)}
+            >
+                {formatMessage(noItemsInBasketMessage)}
+            </Box>
+        )
+    }
+
+    // Loading state
+    if (productsLoading) {
+        return (
+            <Box
+                p={8}
+                textAlign="center"
+                color="gray.500"
+                role="status"
+                aria-live="polite"
+                aria-label={formatMessage(LOADING_LABEL)}
+            >
+                {formatMessage(LOADING_MESSAGE)}
+            </Box>
+        )
+    }
+
+    // Error state
+    if (productsError) {
+        return (
+            <Box
+                p={8}
+                textAlign="center"
+                color="red.500"
+                role="alert"
+                aria-live="assertive"
+                aria-label={formatMessage(ERROR_LABEL)}
+            >
+                {formatMessage(ERROR_MESSAGE)}
+            </Box>
+        )
+    }
+
+    const onAddNewAddress = () => {
+        // TODO: Implement add new address logic/modal
+        console.log('Add New Address clicked!')
+    }
+
+    return (
+        <Box as="main" role="main" aria-label={formatMessage(MAIN_LABEL)}>
```

Comment:
The component is quite large (523 lines). How about we breaking it down?
```
const ProductCard = ({ item, variant, image, addressKey, onAddressChange }) => { ... }
const AddressSelector = ({ addresses, selectedId, onChange, label }) => { ... }

```

Reply:
@alexvuong Sure .. Can we please do this in the next PR? This PR is turning out to be too big.

Reply:
@alexvuong I will create a new WI for this and your other [comment](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2782#discussion_r2216980164)

Reply:
Sorry, the changes in toggle card is important and I will not approve it is not change